Proposals/NNNN-base64-urlencoding-and-omitting-padding.md

```diff
@@ -0,0 +1,83 @@
+# Adding base64 urlencoding and omitting padding option to base64 encoding and decoding
+
+* Proposal: SF-NNNN
+* Authors: [Fabian Fett](https://github.com/fabianfett)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Bug: *if applicable* [apple/swift#NNNNN](https://github.com/apple/swift-foundation/issues/NNNNN)
+* Implementation: [apple/swift-foundation#NNNNN](https://github.com/apple/swift-foundation/pull/NNNNN)
+* Review: ([pitch](https://forums.swift.org/...))
+
+## Revision history
+
+* **v1** Initial version
+
+## Introduction
+
+Introducing base64 encoding and decoding options to support the base64url alphabet as defined in [RFC4648] and to allow the omission of padding characters.
+
+## Motivation
+
+Foundation offers APIs to encode data in the base64 format and to decode base64 encoded data. Multiple RFCs that define cryptography for the web use the base64url encoding and strip the padding characters in the end. Examples for this are: 
+
+- [RFC7519 - JSON Web Token (JWT)][RFC7519]
+- [RFC8291 - Message Encryption for Web Push][RFC8291]
+
+Since Foundation is not offering an API to support the base64url alphabet and omitting the padding characters, users create wrappers around the existing APIs using `replacingOccurrences(of:, with:)`. While this approach works it is very inefficient. The data has to be iterated three times, where one time could have been sufficient.
+
+## Solution
+
+We propose to add additional options to `Data.Base64EncodingOptions`:
+
+```swift
+extension Data.Base64EncodingOptions {
+    /// Use the base64url alphabet to encode the data
+    @available(FoundationPreview 6.2, *)
+    public static var base64UrlAlphabet: Base64EncodingOptions { get }
```

Comment:
Spelling nitpick - I think this should be capitalized like `base64URLAlphabet` (same for others that use URL).